src/components/scenes/CoinRankingScene.tsx

```diff
@@ -0,0 +1,290 @@
+import * as React from 'react'
+import { TouchableOpacity, View } from 'react-native'
+import { FlatList } from 'react-native-gesture-handler'
+
+import { useAsyncEffect } from '../../hooks/useAsyncEffect'
+import { useHandler } from '../../hooks/useHandler'
+import s from '../../locales/strings'
+import { asCoinrankings, CoinRowData, PercentChangeTimeFrame, PriceSubText } from '../../types/coinrankTypes'
+import { useState } from '../../types/reactHooks'
+import { FlatListItem } from '../../types/types'
+import { debugLog, enableDebugLogType, LOG_COINRANK } from '../../util/logger'
+import { fetchRates } from '../../util/network'
+import { SceneWrapper } from '../common/SceneWrapper'
+import { CoinRankRow } from '../data/row/CoinRankRow'
+import { showError } from '../services/AirshipInstance'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { DividerLine } from '../themed/DividerLine'
+import { EdgeText } from '../themed/EdgeText'
+import { OutlinedTextInput, OutlinedTextInputRef } from '../themed/OutlinedTextInput'
+
+const coinRowData: CoinRowData = { coinRankings: [] }
+
+const QUERY_PAGE_SIZE = 30
+const LISTINGS_REFRESH_INTERVAL = 30000
+
+// Masking enable bit with 0 disables logging
+enableDebugLogType(LOG_COINRANK & 0)
+
+interface Props {
+  // navigation: NavigationProp<'coinRanking'>
+  // route: RouteProp<'coinRanking'>
+}
+
+const percentChangeOrder: PercentChangeTimeFrame[] = ['hours1', 'hours24', 'days7', 'days30', 'year1']
+const percentChangeStrings: { [pc: string]: string } = {
+  hours1: '1hr',
+  hours24: '24hr',
+  days7: '7d',
+  days30: '30d',
+  year1: '1y'
+}
+const priceSubTextStrings: { [pc: string]: string } = {
+  marketCap: 'MCap',
+  volume24h: 'Vol 24h'
```

Comment:
Do we need to add these to enus for translation?

Reply:
Let's give it a shot. I'm afraid translators will botch this.